src/Cake.Common/Tools/NSIS/MakeNSISRunner.cs

```diff
@@ -102,7 +102,7 @@ private ProcessArgumentBuilder GetArguments(FilePath scriptFile, MakeNSISSetting
             }
 
             // Script file
-            builder.Append(scriptFile.MakeAbsolute(_environment).FullPath);
+            builder.Append("\"" + scriptFile.MakeAbsolute(_environment).FullPath + "\"");
```

Comment:
Would prefer if this was changed to `builder.AppendQuoted(scriptFile.MakeAbsolute(_environment).FullPath)`.